pkg/server/sandbox_run.go

```diff
@@ -553,3 +525,48 @@ func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []cni.PortMapping
 	}
 	return portMappings
 }
+
+// untrustedWorkload returns true if the sandbox contains untrusted workload.
+func untrustedWorkload(config *runtime.PodSandboxConfig) bool {
+	return config.GetAnnotations()[annotations.UntrustedWorkload] == "true"
+}
+
+// hostPrivilegedSandbox returns true if the sandbox configuration
+// requires additional host privileges for the sandbox.
+func hostPrivilegedSandbox(config *runtime.PodSandboxConfig) bool {
+	securityContext := config.GetLinux().GetSecurityContext()
+	if securityContext.GetPrivileged() {
+		return true
+	}
+
+	namespaceOptions := securityContext.GetNamespaceOptions()
+	if namespaceOptions.GetNetwork() == runtime.NamespaceMode_NODE ||
+		namespaceOptions.GetPid() == runtime.NamespaceMode_NODE ||
+		namespaceOptions.GetIpc() == runtime.NamespaceMode_NODE {
+		return true
+	}
+
+	return false
+}
+
+// getSandboxRuntime returns the runtime configuration for sandbox.
+// If the sandbox contains untrusted workload, secure runtime will be returned,
+// or else default runtime will be returned.
+func (c *criService) getSandboxRuntime(config *runtime.PodSandboxConfig) (criconfig.Runtime, error) {
+	secureRuntimeRequired := false
+	if untrustedWorkload(config) {
```

Comment:
untrustedWorkload will check for UntrustedWorkload being set. But because the sandbox is not created, this was never set before (is not set by k8s today). 
Here we need to  check:
```
if is a *not* a privileged sandbox;then
set UntrustedWorkload
set secureRuntimeRequired
fi
```

```diff
+func (c *criService) getSandboxRuntime(config *runtime.PodSandboxConfig, sandbox *sandboxstore.Sandbox) (criconfig.Runtime, error) {
        secureRuntimeRequired := false
-       if untrustedWorkload(config) {
-               // TODO(random-liu): Figure out we should return error or not.
-               if hostPrivilegedSandbox(config) {
-                       return criconfig.Runtime{}, errors.New("untrusted workload with host privilege is not allowed")
-               }
+       if !hostPrivilegedSandbox(config) {
+               sandbox.Config.Annotations[annotations.UntrustedWorkload] = "true"
                secureRuntimeRequired = true
        }
 ```

Reply:
@jcvenegas Can we chat? I think I may have misunderstood something here.

Reply:
As an example, if user create a pod like this:
```
apiVersion: v1
kind: Pod
metadata:
  name: nginx
  annotations:
    io.kubernetes.cri.untrusted-workload: true
spec:
  containers:
  - name: nginx
    image: nginx
```

We'll run it in kata.